src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,72 @@
+import React from 'react';
+import { Link, useLocation, useSearchParams } from 'react-router-dom';
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const availableCenturies = [16, 17, 18, 19, 20];
+
+  const pathName = useLocation().pathname;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const searchInputValue = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+
+  function setParamsWith(params: SearchParams) {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+
+    return search;
+  }
+
+  function handleSearchInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const query = event.target.value === '' ? null : event.target.value;
+
+    const search = setParamsWith({ query });
+
+    setSearchParams(search);
+  }
+
+  function handleAddNewCentury(century: number) {
+    const newCenturies = centuries.includes(century.toString())
+      ? centuries.filter(c => c !== century.toString())
+      : [...centuries, century.toString()];
+
+    return newCenturies;
```

Comment:
The `handleAddNewCentury` function returns `newCenturies`, but this value is not used directly in the component. Instead, it's passed to `getSearchWith` in the `to` prop of the `Link` component. Ensure that the logic for updating the centuries is correctly handled in the `getSearchWith` function and that the returned value is used appropriately.